Data Domain 모듈을 분리하여 프로젝트 생성하기

ParkChan opened the pull request:
요구 사항
순수 코틀린 모듈인 domain 모듈을 만든다.
순수 코틀린 모듈인 data 모듈을 만든다.
data 모듈은 domain 모듈을 implementation 해야 한다.
app 모듈은 data, domain 모듈을 implementation 해야 한다.

프로젝트를 생성하여 모듈 구성을 하였습니다.

ParkChan committed:
[추가] data, domain 모듈 만들기 #1

요구 사항
순수 코틀린 모듈인 domain 모듈을 만든다.
순수 코틀린 모듈인 data 모듈을 만든다.
data 모듈은 domain 모듈을 implementation 해야 한다.
app 모듈은 data, domain 모듈을 implementation 해야 한다.

ParkChan committed:
[추가] 멀티 모듈들의 gradle 의존성 관리를 위한 buildSrc

BeokBeok commented on buildSrc/src/main/java/Dependencies.kt:
```diff
@@ -0,0 +1,98 @@
+object RootConfig {
+    const val GRADLE = "com.android.tools.build:gradle:7.0.2"
+    const val KOTLIN_GRADLE_PLUGIN = "org.jetbrains.kotlin:kotlin-gradle-plugin:${Kotlin.VERSION}"
+    const val DAGGER_HILT_PLUGIN = "com.google.dagger:hilt-android-gradle-plugin:2.28-alpha"
```
P2. 힐트 버전을 참조하게 해서 힐트 버전이 변경되면 이 부분도 같이 변경되면 좋을 것 같아요

ParkChan replied:
653d6a957925c347363fd5244365fd8245b72b1b 버전 참조를 같이 하도록 수정하였습니다.

BeokBeok commented on buildSrc/src/main/java/Dependencies.kt:
```diff
@@ -0,0 +1,98 @@
+    const val LIFECYCLE_VIEW_MODEL = "androidx.lifecycle:lifecycle-viewmodel-ktx:$LIFECYCLE_VERSION"
+    const val LIFECYCLE_RUNTIME = "androidx.lifecycle:lifecycle-runtime-ktx:$LIFECYCLE_VERSION"
+    const val LIFECYCLE_LIVEDATA = "androidx.lifecycle:lifecycle-livedata-ktx:$LIFECYCLE_VERSION"
+    const val LIFECYCLE_EXT = "androidx.lifecycle:lifecycle-extensions:$LIFECYCLE_VERSION"
```
P2. lifecycle-extensions 는 deprecated 되었어요.
지워도 될 것 같아요

ParkChan replied:
71927ec365d328d7a06133a68d13dbb2eb1ab0c3 deprecated 된 항목으로 삭제 하였습니다.

BeokBeok commented on buildSrc/src/main/java/Dependencies.kt:
```diff
@@ -0,0 +1,98 @@
+    const val CONSTRAINT_LAYOUT = "androidx.constraintlayout:constraintlayout:1.1.3"
+
+    //ActivityResultContract
+    const val ACTIVITY_KTX = "androidx.activity:activity-ktx:1.2.0-alpha06"
+    const val FRAGMENT_KTX = "androidx.fragment:fragment-ktx:1.3.0-alpha06"
```
P2. 정식 릴리즈 버전을 사용해주시면 좋을 것 같아요

ParkChan replied:
e0894a1a85d7518fd1d987b4ff1c05a618cb06a7 알려주신 maven repository를 통해 
정식 릴리즈 버전을 참고하여 수정하였습니다.

https://mvnrepository.com/artifact/androidx.activity/activity-ktx
https://mvnrepository.com/artifact/androidx.fragment/fragment-ktx

BeokBeok commented on data/build.gradle:
```diff
@@ -0,0 +1,13 @@
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_7
+    targetCompatibility = JavaVersion.VERSION_1_7
```
P2. 버전을 1_8로 변경해주세요

ParkChan replied:
456335dcabc7e1517b6f8d1d102346ff5cf219d9 수정하였습니다.

BeokBeok commented on domain/build.gradle:
```diff
@@ -0,0 +1,9 @@
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_7
+    targetCompatibility = JavaVersion.VERSION_1_7
```
P2. 버전을 1_8로 변경해주세요

ParkChan replied:
c2e62a0e85e231240d8dd485425fbe5d19f9b13a 변경하였습니다.

BeokBeok reviewed (commented):
작업하시느라 고생이 많으십니다.
약간의 코맨트를 작성하였습니다. :smile:

<details><summary>코드리뷰 우선순위 규칙 (P1~P3)</summary>
P1: 적극적으로 고민해주세요 (Request changes)
<br>
코드 수정이 반드시 필요하다고 판단되는 경우입니다.<br>
리뷰 요청자께서 만약 수용할 수 없는 상황이라면 적합한 의견을 들어 토론해주세요.
<br></br>
P2: 웬만하면 반영해 주세요 (Comment)
<br>
코드 수정을 권장하거나 리뷰 요청자께서 해당 코드를 작성하신 의도를 알고자 하는 경우입니다.<br>
리뷰 요청자께서는 가급적이면 이유를 들어 설명해주시거나 의견을 달아주세요.
<br></br>
P3: 반영해도 좋고 넘어가도 좋습니다 (Approve)
<br>
해당 의견을 반영하는 게 좋을지 고민해 보는 정도면 충분합니다.<br>
만약 리뷰 요청자께서 리뷰어의 의견을 반영하셨더라도 리뷰어에게 별도로 리뷰 요청을 드리지 않아도 됩니다.
</details>

BeokBeok commented:
@ParkChan 
PR 이름을 보았을 때, 해당 PR이 어떤 역할을 하는지 알 수 있을까요?

BeokBeok commented:
@ParkChan 
feature 브랜치에서 develop 브랜치로 머지해주세요

ParkChan committed:
[수정] root상의 .idea 삭제 및 gitinore 등록

ParkChan committed:
[PR/수정] P2. 힐트 버전을 참조하게 해서 힐트 버전이 변경되면 이 부분도 같이 변경되면 좋을 것 같아요

ParkChan committed:
[PR/수정] P2. lifecycle-extensions 는 deprecated 되었어요. 지워도 될 것 같아요

삭제

ParkChan committed:
[PR/수정] P2. 정식 릴리즈 버전을 사용해주시면 좋을 것 같아요

maven repository 버전 확인을 통해 정식 릴리즈 버전으로 수정함
https://mvnrepository.com/artifact/androidx.activity/activity-ktx
https://mvnrepository.com/artifact/androidx.fragment/fragment-ktx

ParkChan committed:
[PR/수정] P2. 버전을 1_8로 변경해주세요

ParkChan committed:
[PR/수정] P2. 버전을 1_8로 변경해주세요

ParkChan commented:
> @ParkChan PR 이름을 보았을 때, 해당 PR이 어떤 역할을 하는지 알 수 있을까요?

아래와 같이 충분한 설명이 있어 좋은것 같습니다.
예전 P5로 작성하여주신것보다 줄어들어 편해진것 같아요.
문득 드는생각인데 별표시 같은걸로 노출 하면 어떨까요? 
P1 > ★★★ : 적극적으로 고민해주세요 (Request changes)
P2 > ★★☆ : 웬만하면 반영해 주세요 (Comment)
P3 > ★☆☆ : 반영해도 좋고 넘어가도 좋습니다 (Approve)

<details><summary>코드리뷰 우선순위 규칙 (P1~P3)</summary>
P1: 적극적으로 고민해주세요 (Request changes)
<br>
코드 수정이 반드시 필요하다고 판단되는 경우입니다.<br>
리뷰 요청자께서 만약 수용할 수 없는 상황이라면 적합한 의견을 들어 토론해주세요.
<br></br>
P2: 웬만하면 반영해 주세요 (Comment)
<br>
코드 수정을 권장하거나 리뷰 요청자께서 해당 코드를 작성하신 의도를 알고자 하는 경우입니다.<br>
리뷰 요청자께서는 가급적이면 이유를 들어 설명해주시거나 의견을 달아주세요.
<br></br>
P3: 반영해도 좋고 넘어가도 좋습니다 (Approve)
<br>
해당 의견을 반영하는 게 좋을지 고민해 보는 정도면 충분합니다.<br>
만약 리뷰 요청자께서 리뷰어의 의견을 반영하셨더라도 리뷰어에게 별도로 리뷰 요청을 드리지 않아도 됩니다.
</details>

BeokBeok commented:
현재 PR 이름이 Develop 으로 되어 있는데, 다른 개발자가 봐도 어떤 작업을 했는지 알기 쉽게 변경하는 것이 좋습니다.

BeokBeok commented:
> 아래와 같이 충분한 설명이 있어 좋은것 같습니다.
예전 P5로 작성하여주신것보다 줄어들어 편해진것 같아요.
문득 드는생각인데 별표시 같은걸로 노출 하면 어떨까요?
P1 > ★★★ : 적극적으로 고민해주세요 (Request changes)
P2 > ★★☆ : 웬만하면 반영해 주세요 (Comment)
P3 > ★☆☆ : 반영해도 좋고 넘어가도 좋습니다 (Approve)

좋은 제안 감사합니다 👍

BeokBeok commented on buildSrc/src/main/java/Dependencies.kt:
```diff
@@ -0,0 +1,98 @@
+
+    //dagger-hilt
+    const val ANDROID = "com.google.dagger:hilt-android:$HILT_VER"
+    const val ANDROID_COMPILER = "com.google.dagger:hilt-android-compiler:$HILT_VER"
```
P2. hilt-android-compiler 대신 hilt-compiler를 사용해주세요

ParkChan replied:
329ae848e1036ebcb227bf92742f5dcd0fa5f400  hilt-compiler를 사용하도록 수정하였습니다.

BeokBeok commented on buildSrc/src/main/java/Dependencies.kt:
```diff
@@ -0,0 +1,98 @@
+    //dagger-hilt
+    const val ANDROID = "com.google.dagger:hilt-android:$HILT_VER"
+    const val ANDROID_COMPILER = "com.google.dagger:hilt-android-compiler:$HILT_VER"
+    const val LIFECYCLE_VM = "androidx.hilt:hilt-lifecycle-viewmodel:$HILT_JETPACK_VER"
```
P2. 힐트가 버전이 올라가면서 jetpack은 사용하지 않아도 됩니다.

ParkChan replied:
b28a90d97d4d7679db67f2458340776f5d6fd2c7 jetpack은 사용하지 않도록 삭제하였습니다. 
추가적으로 낮은 버전을 사용하는 몇몇 라이브러리들을 최신 버전으로 변경 하였습니다.

ParkChan committed:
[PR/수정] P2. hilt-android-compiler 대신 hilt-compiler를 사용해주세요

ParkChan committed:
[PR/수정] P2. 힐트가 버전이 올라가면서 jetpack은 사용하지 않아도 됩니다.

BeokBeok commented on buildSrc/src/main/java/Dependencies.kt:
```diff
@@ -0,0 +1,95 @@
+
+    //dagger-hilt
+    const val ANDROID = "com.google.dagger:hilt-android:$HILT_VER"
+    const val ANDROID_COMPILER = "com.google.dagger:hilt-compiler:$HILT_VER"
```
P3. 이름도 COMPILER로 변경되면 좋을 것 같아요

ParkChan replied:
59f568e4f7b68db0ae47206d9f92a1c1f2db3919 변경하였습니다.

BeokBeok reviewed (dismissed):
<img src="https://user-images.githubusercontent.com/48344355/108810482-8fca5400-75ee-11eb-9a24-9931a3877f9d.png" width="30%">

ParkChan committed:
P3. 이름도 COMPILER로 변경되면 좋을 것 같아요
